cmd/osmosisd/cmd/root.go

```diff
@@ -401,6 +417,52 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 	return rootCmd, encodingConfig
 }
 
+// overwrites config.toml values if it exists, otherwise it writes the default config.toml
+func overwriteConfigTomlValues(serverCtx *server.Context) (*tmcfg.Config, error) {
+	// Get paths to config.toml and config parent directory
+	rootDir := serverCtx.Viper.GetString(flags.FlagHome)
+	configParentDirPath := filepath.Join(rootDir, "config")
+	configFilePath := filepath.Join(configParentDirPath, "config.toml")
+
+	// Initialize default config
+	tmcConfig := tmcfg.DefaultConfig()
+
+	_, err := os.Stat(configFilePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to read in %s: %w", configFilePath, err)
+		}
+
+		// If does not exist, write default config.toml to update
+
+		// We modify the default config.toml to have faster block times
+		// It will be written by server.InterceptConfigsPreRunHandler
+		tmcConfig.Consensus.TimeoutCommit = 4 * time.Second
+	} else {
+		// config.toml exists
+
+		serverCtx.Viper.SetConfigType("toml")
+		serverCtx.Viper.SetConfigName("config")
+		serverCtx.Viper.AddConfigPath(configParentDirPath)
+
+		// We read it in and modify the consensus timeout commit
+		// and write it back.
+		if err := serverCtx.Viper.ReadInConfig(); err != nil {
+			return nil, fmt.Errorf("failed to read in %s: %w", configFilePath, err)
+		}
+
+		// The original default is 5s and is set in Cosmos SDK.
+		// We lower it to 4s for faster block times.
+		serverCtx.Config.Consensus.TimeoutCommit = 4 * time.Second
+
+		// It will be re-read in server.InterceptConfigsPreRunHandler
+		tmcfg.WriteConfigFile(configFilePath, serverCtx.Config)
+
+		tmcConfig = serverCtx.Config
+	}
+	return tmcConfig, nil
+}
```

Comment:
Nice! I am told that there is a more canonical way to do these initializations in SDK 47, but we can write issue as a future todo for ouselves on this